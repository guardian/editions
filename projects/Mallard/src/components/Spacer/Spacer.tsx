import React from 'react';
import { StyleSheet, View } from 'react-native';

const styles = (unit: number) =>
	StyleSheet.create({
		spacer: {
			marginRight: unit,
			marginTop: unit,
		},
	});
export const Spacer = ({ unit = 8 }: { unit?: number }) => (
	<View style={styles(unit).spacer} />
);
