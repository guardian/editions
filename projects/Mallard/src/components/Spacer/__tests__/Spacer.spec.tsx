import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Spacer } from '../Spacer';

describe('<Spacer />', () => {
	it('should render with default props', () => {
		const wrapper = TestRenderer.create(<Spacer />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});

	it('should render with given props', () => {
		const wrapper = TestRenderer.create(<Spacer unit={30} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
