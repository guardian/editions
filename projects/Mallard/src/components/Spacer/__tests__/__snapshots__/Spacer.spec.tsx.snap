// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacer /> should render with default props 1`] = `
<View
  style={
    {
      "marginRight": 8,
      "marginTop": 8,
    }
  }
/>
`;

exports[`<Spacer /> should render with given props 1`] = `
<View
  style={
    {
      "marginRight": 30,
      "marginTop": 30,
    }
  }
/>
`;
